gnome-initial-setup/demo/eos-demo-video.c

```diff
@@ -132,14 +137,32 @@ gis_demo_video_init (GisDemoVideo *demo_video)
   GisDemoVideoPrivate *priv = gis_demo_video_get_instance_private (demo_video);
   GstBus *bus = NULL;
 
+  const gchar * const *languages;
+  gchar file_path[PATH_MAX];
+  gchar file_uri[PATH_MAX];
+  int i;
+  
   priv->element = gst_element_factory_make ("gtksink", NULL);
   g_object_get (&priv->element->object, "widget", &priv->widget, NULL);
 
   priv->player = gst_element_factory_make ("playbin", "GisDemoVideoPlayBin");
   priv->bus = gst_pipeline_get_bus (GST_PIPELINE (priv->player));
 
   g_object_set (priv->player, "video-sink", g_object_ref_sink (priv->element), NULL);
-  g_object_set (priv->player, "uri", DEMO_VIDEO_FILE_URI, NULL);
+
+  languages = g_get_language_names ();
+
+  for (i = 0; languages[i] != NULL; i++)
+    {
+      g_snprintf(file_path, PATH_MAX, DEMO_VIDEO_FILE_PATH, languages[i]);
+      if (g_file_test (file_path, G_FILE_TEST_EXISTS))
+        {
+	  g_snprintf(file_uri, PATH_MAX, DEMO_VIDEO_FILE_URI, languages[i]);
```

Comment:
This may not always yield what you want as you're using the same `PATH_MAX` for both the `file_path` and the `file_uri` but the latter has a longer prefix, so if the `file_path` reached the `PATH_MAX` chars minus the length of `file://`, it will result in an invalid `file_uri` here. I know this is unlikely to happen but we should still make sure there's no possibility of it ever happening.

I would simplify this by keeping only the `file_uri`, `g_snprintf` it above where you're doing it to the `file_path`, but then just check for the file existence by doing `file_path + 7` (so it checks it from after the `file://` prefix). Alternatively you can build a `GFile` with the URI but that involves allocating memory which would be overkill for such a small check IMO.

Reply:
Good idea.  I'll go with the simpler approach.